reference/README.md

```diff
@@ -6,7 +6,7 @@ The C concept exercises are based on language concepts. The list below contains
 
 ### Specialities
 
-#### Concepts
+#### Speciality Concepts
```

Comment:
Do you want "specialty" here?

Reply:
Because there are two headings at the H3 level, Specialities and General, and these each have matching sub-headings (i.e. at the H4 level), it means there are two `Concept` headings and two `Types` headings. 

This runs afoul of [MD024/no-duplicate-heading/no-duplicate-header](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md024---multiple-headings-with-the-same-content)

Prefixing the H4 headings seemed like a quick fix that keeps the meaning clear and explicit.

Reply:
Ack, UK-Eng  vs US-Eng strikes again!

Reply:
Hmm, where are you seeing that it's specialities in the US? I've never seen that in use, but it may be a regional thing.